File: web/di_core/DataIngestion/Domain/Response/JdbcSource.ts
/*
 * @author: tvc12 - Thien Vi
 * @created: 6/1/21, 2:17 PM
 */

import { DataSourceType, DataSources } from '@core/DataIngestion';
import { DataSource } from '@core/DataIngestion/Domain/Response/DataSource';
import { SourceId } from '@core/domain';

export class JdbcSource implements DataSource {
  id: SourceId;
  orgId: string;
  databaseType: DataSourceType;
  displayName: string;
  jdbcUrl: string;
  username: string;
  password: string;
  lastModify: number;
  readonly className = DataSources.JdbcSource;

  constructor(
    id: SourceId,
    orgId: string,
    databaseType: DataSourceType,
    displayName: string,
    jdbcURL: string,
    username: string,
    password: string,
    lastModify: number
  ) {
    this.id = id;
    this.orgId = orgId;
    this.displayName = displayName;
    this.databaseType = databaseType;
    this.jdbcUrl = jdbcURL;
    this.username = username;
    this.password = password;
    this.lastModify = lastModify;
  }

  static fromObject(obj: any): JdbcSource {
    return new JdbcSource(obj.id, obj.orgId, obj.databaseType, obj.displayName, obj.jdbcUrl, obj.username, obj.password, obj.lastModify);
  }
}

File: rocket-bi-server/src/main/scala/co/datainsider/jobworker/domain/source/JdbcSource.scala
package co.datainsider.jobworker.domain.source

import com.fasterxml.jackson.module.scala.JsonScalaEnumeration
import co.datainsider.jobworker.domain.DataSourceType.DataSourceType
import co.datainsider.jobworker.domain.DatabaseType.DatabaseType
import co.datainsider.jobworker.domain.{DataSource, DataSourceType, DatabaseTypeRef}
import co.datainsider.jobworker.domain.Ids.SourceId

case class JdbcSource(
                       orgId: Long = -1,
                       id: SourceId,
                       displayName: String,
                       @JsonScalaEnumeration(classOf[DatabaseTypeRef]) databaseType: DatabaseType,
                       jdbcUrl: String,
                       username: String,
                       password: String
                     ) extends DataSource {

  override def getId: SourceId = id

  override def getName: String = displayName

  override def getType: DataSourceType = DataSourceType.Jdbc

  override def getConfig: Map[String, Any] = {
    Map("database_type" -> databaseType.toString, "jdbc_url" -> jdbcUrl, "username" -> username, "password" -> password)
  }

}
